src/main/java/hu/elte/eserial/mapper/ArrayMapper.java

```diff
@@ -11,26 +12,34 @@
 /**
  * Maps array-like objects (e.g Integer[]).
  */
-public class ArrayMapper implements ObjectMapper {
+public class ArrayMapper extends AbstractMapper {
 
     /**
-     * Returns the mapped representation of the array {@code that}.
+     * Constructs an {@link ArrayMapper} and sets the {@code object} in it.
      *
-     * @param {@code that} an array
-     * @return mapped representation of {@code that}
+     * @param object the {@code array} to be used in the {@link AbstractMapper#map} method
+     */
+    ArrayMapper(Object object) {
+        super(object);
+    }
+
+    /**
+     * @param recursionChecker {@inheritDoc}
+     * @return mapped representation of the contained array
      */
     @Override
-    public Object map(Object that) {
-        if (!TypeUtils.isArray(that.getClass())) {
-            throw new EserialMapperMismatchException(Array.class.getSimpleName(), that.getClass().getSimpleName());
+    public Object map(RecursionChecker recursionChecker) {
+        if (!TypeUtils.isArray(this.object.getClass())) {
```

Comment:
What if object is null? Applies to the other mappers as well.

Reply:
We aren't really consequent with the usage of `NullPointerException`s vs meaningful `EserialException`s. Several unit tests depend on having an NPE thrown.

This inconsistency should be handled in a meeting!